libnavigation-core/src/main/java/com/mapbox/navigation/core/internal/fasterroute/RecordRouteObserverResults.kt

```diff
@@ -0,0 +1,100 @@
+package com.mapbox.navigation.core.internal.fasterroute
+
+import com.google.gson.GsonBuilder
+import com.google.gson.TypeAdapter
+import com.google.gson.stream.JsonReader
+import com.google.gson.stream.JsonWriter
+import com.mapbox.navigation.base.route.NavigationRoute
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.directions.session.RoutesObserver
+import com.mapbox.navigation.core.directions.session.RoutesUpdatedResult
+import com.mapbox.navigation.core.routealternatives.AlternativeRouteMetadata
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Deferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.async
+import kotlinx.coroutines.launch
+import java.io.File
+
+/***
+ * Records routes updates to record-routes-observer folder.
+ * You can copy records from device file system to unit test resources and replay them using
+ * com.mapbox.navigation.core.fasterroute.ReplayRouteObserverResultsKt.readRouteObserverResults
+ */
+class RecordRouteObserverResults(
+    private val navigation: () -> MapboxNavigation
```

Comment:
`RecordRouteObserverResults` is a simple tool for testing of one feature. I don't expect it to be used by anyone else. I'm not sure if it's worth investing time in this instrument. Do you think it can be useful outside of faster route feature?

Reply:
Oh, I didn't notice it was internal.
Anyway, internal API can be used in examples, 1TAP or Copilot. I'd vote for making the instrument simpler to make our own lives easier. But it's not that a big a deal as it would be if it were public API.